lib/interpreters/split_clustering.rb

```diff
@@ -122,15 +122,298 @@ def self.generate_split_vrps(service_vrp, _job = nil, block = nil)
+      points = []
+      services = []
+      # TODO: 0- relations needs to be taken into account inside clustering during this split
+      #          - the vehicles need to be in the same sub-vrp for the following relations:
+      #            => vehicle_trips
+      #          - the services need to be in the same sub-vrp for the following relations:
+      #            => meetup, minimum_duration_lapse, maximum_duration_lapse, minimum_day_lapse, maximum_day_lapse
+      #            (we need to go thorugh the original relations and "connect" the "vehicle_id"s below with "same_route")
+      relations = []
+
+      split_solve_data[:service_vehicle_assignments].each{ |vehicle_id, vehicle_services|
+        # TODO: After relations are taken into account inside clustering, we don't have to
+        #       decrease the number of points to 1. We can represent each group with multiple
+        #       points, carefully selected to represent the mean, median and extremes of the group
+        #       and "relate" these points so that they will stay on the same "side" in the 2-split
+        average_lat = vehicle_services.sum{ |s| s.activity.point.location.lat } / vehicle_services.size.to_f
+        average_lon = vehicle_services.sum{ |s| s.activity.point.location.lon } / vehicle_services.size.to_f
+        points << { id: "p#{vehicle_id}", location: { lat: average_lat, lon: average_lon }}
+        services << {
+          id: vehicle_id, # vehicle_id used to find the original service-vehicle assignment
+          visits_number: vehicle_services.size,
+          activity: {
+            point_id: "p#{vehicle_id}",
+            duration: vehicle_services.sum{ |s| s.activity.duration.to_f } / Math.sqrt(vehicle_services.size)
+          }
+        }
+      }
+
+      # TODO: The following two "fake" vehicles can have carefully selected start and end points!
+      #       So that if there are multiple zone/cities or multiple depots, the split will be
+      #       more intelligent. For that we need go over the list of uniq depots and select two
+      #       depots from the list that "split" the depots into two groups and minimize the total
+      #       distance between the selected depots. Then these two depots can be used as the
+      #       depots for these two "fake" vehicles.
+      ::Models::Vrp.create({
+        name: 'SPLIT_VRP',
+        points: points,
+        vehicles: Array.new(2){ |i| { id: "v#{i}", router_mode: 'car' } },
+        services: services,
+        relations: relations
+      }, false)
+    end
+
+    def self.create_fake_sub_split_vrp(split_solve_data)
+      split_vrp = split_solve_data[:split_vrp]
+      sub_split_vrp = ::Models::Vrp.create({ name: 'SUB_SPLIT_VRP' }, false)
+      sub_split_vrp.vehicles = split_vrp.vehicles # 2-fake-vehicles
+      sub_split_vrp.services = split_vrp.services.select{ |service|
+        split_solve_data[:current_vehicles].any?{ |v| v.id == service.id }
```

Comment:
This line is confusing to me. I just want to make sure this is not a bug, but I still need to investigate more to make sure split_solve_data[:current_vehicles] contain service.ids.

Reply:
I feel like split_solve_data[:current_vehicles] is define by  ```current_vehicles: vrp.vehicles.map{ |v| v }```